### PR TITLE
[PERFORMANCE] Pickle5 ConnectionWrapper for VectorEnv at least (2-5% speed up)

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context

* [PEP 574](https://www.python.org/dev/peps/pep-0574/) introduced a zero-copy pickling method that makes the pickling of Numpy arrays significantly faster. However, this feature is only supported on the latest Pickle protocol (5) which is not used by the default for multiprocessing (or even torch.multiprocessing, although it is often used in wrappers like Queues). 

I ported and slightly modified the torch Connection wrapper code to habitat-lab and improved it by adding a pickle5 backport. The backport port brings this functionality back to Python<3.8 After using this PR, it should remove a copy from NumPy arrays.

Furthermore, this should not affect the pickling of any other parts of the code stack or any other multiprocessing code.

This speed up is likely to be a lot more on higher resolution outputs from habitat-sim as well as the sensor numbers increase.

This change should be fully backwards compatible and speed things up across the board without any code changes for the end user.

## How Has This Been Tested
Local testing shows at least 2-5% speed up in VectorEnv evaluation.  The entire test suite appears now takes 10 seconds less (despite only a fraction of the tests actually using VectorEnv.) I have tested this in Python 3.8 as well as Python 3.6.

As an aside, we should use add tests for pytest-benchmarking on common habitat-lab components like VectorEnv.

## Types of changes
- Docs change / refactoring / dependency upgrade

## Checklist

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by erikwijmans** [comment]
This is great!  I think we could push this further by using CPU shared memory as that would also allow observations, rewards, etc to be batched by default instead of having to batch in habitat_baselines, but that can be done later

File: habitat/utils/pickle5_multiprocessing.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import sys
+from multiprocessing.connection import Connection
+from multiprocessing.reduction import ForkingPickler as _ForkingPickler
+
+if sys.version_info[:2] < (3, 8):
+    # pickle 5 backport
+    try:
+        import pickle5 as pickle
+    except ImportError:
+        import pickle  # type: ignore[no-redef]
```
Comment (Skylion007) [inline habitat/utils/pickle5_multiprocessing.py:17]: @erikwijmans Only thing I am unsure is what to log here: Should we give a warning? And should we use the `habitat.core.logger` or `warning.warn`

Comment (erikwijmans) [reply habitat/utils/pickle5_multiprocessing.py:17]: Probably logger.  This will only run once anyways

**Review by mathfac — APPROVED** [review]
That's amazing. Thank you for backward compatibility. Let's merge it.

**Pull request merged**